scripts/check-docs-quality.js

```diff
@@ -16,53 +16,67 @@
  */
 const { execSync, spawnSync } = require('child_process');
 const path = require('path');
+// eslint-disable-next-line import/no-extraneous-dependencies
+const commandExists = require('command-exists');
 
 const listFilesTrackedByGit = 'git ls-files';
 
 const inheritStdIo = {
   stdio: 'inherit',
 };
 
-const ERROR_MESSAGE =
-  'Please install vale linter(https://docs.errata.ai/vale/install). Ignore this message if already installed.\n';
+const LINT_SKIPPED_MESSAGE =
+  'Skipping documentation quality check (vale not found). Install vale linter (https://docs.errata.ai/vale/install) to enable.\n';
+const ERROR_MESSAGE = `Language linter (vale) generated errors. Please check the errors and review any markdown files that you changed.
+  Possibly update .github/styles/vocab.txt to add new valid words.\n`;
 
-// xargs is not supported by shx.
-if (process.platform === 'win32') {
-  const validMDFilesCommand = `${listFilesTrackedByGit} | .\\node_modules\\.bin\\shx grep ".md"`;
-  try {
-    // get list of all md files except in directories of gitignore.
-    let filesToLint = execSync(validMDFilesCommand, {
-      stdio: ['ignore', 'pipe', 'inherit'],
-    });
+// Proceed with the script only if Vale linter is installed. Limit the friction and surprises caused by the script.
+commandExists('vale')
```

Comment:
Probably makes sense to keep the `lint-staged` logic behind a flag, or the other way around. We wanna make sure that `vale` is always present in CI, both for us and whoever might reuse this script.

Reply:
Makes sense.

The script can check if `CI` env variable is set to true. If yes, the script will make sure `vale` is installed. If false, the script assumes to be run by `lint-staged` and does not cause errors if `vale` is not installed.

Ref: `CI` is always set to true on GitHub CI https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables

Reply:
Done!